source/codegen/templates/service_helpers.mako

```diff
@@ -565,7 +575,11 @@ ${initialize_standard_input_param(function_name, parameter)}
   c_element_type_that_needs_coercion = service_helpers.get_c_element_type_for_array_that_needs_coercion(parameter)
 %>\
 % if c_type in ['ViConstString', 'const char[]']:
-      auto ${parameter_name} = ${request_snippet}.c_str();\
+      auto ${parameter_name}_mbcs = convert_from_grpc<std::string>(${request_snippet});
+      auto ${parameter_name} = ${parameter_name}_mbcs.c_str();\
+% elif grpc_type == 'string':
+      auto ${parameter_name}_mbcs = convert_from_grpc<std::string>(${request_snippet});
+      ${c_type_pointer} ${parameter_name} = (${c_type_pointer})${parameter_name}_mbcs.c_str();\
```

Comment:
Above on line 463 in a similar case you're doing `const_cast` - should these be consistent? It would seem more semantically correct than a C-style cast.

Reply:
1. In addition to const_cast, it would also need a static_cast or reinterpret_cast, since the c_type_pointer is not a `char*`.
2. I made it this way to match the nearby code, 2 lines down.